src/main/java/org/folio/service/DefaultRecordLoaderService.java

```diff
@@ -0,0 +1,56 @@
+package org.folio.service;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.folio.domain.Batch;
+import org.folio.domain.records.FolioRecord;
+import org.folio.domain.records.Instance;
+
+public class DefaultRecordLoaderService implements RecordLoaderService {
+
+
+  private final FileUploadService fileUploadService;
+
+  public DefaultRecordLoaderService(FileUploadService fileUploadService) {
+    this.fileUploadService = fileUploadService;
+  }
+
+  @Override
+  public Stream<Batch<? extends FolioRecord>> getBatchRecordsStream(int batchSize) throws IOException {
+    Stream<List<String>> sourceStream = fileUploadService.getSourceStream(batchSize);
+    return sourceStream
+      .map(this::callSrs)
```

Comment:
As we work in the async world the method callSrs returns Future. Here we can not rely on async call in sync method.